system/src/androidLibMain/kotlin/com/splendo/kaluga/system/network/NetworkManager.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import android.content.Context
+import android.content.IntentFilter
+import android.net.ConnectivityManager
+import android.os.Build
+import co.touchlab.stately.concurrency.AtomicReference
+import com.splendo.kaluga.system.network.services.ConnectivityCallbackNetworkManager
+import com.splendo.kaluga.system.network.services.ConnectivityReceiverNetworkManager
+import com.splendo.kaluga.system.network.services.NetworkManagerService
+
+interface NetworkHelper {
+    fun determineNetworkType(): Network
+}
+
+actual class NetworkManager actual constructor(
+    networkStateRepo: NetworkStateRepo,
+    private val context: Any?
+) : BaseNetworkManager(networkStateRepo), NetworkManagerService {
+
+    private val connectivityManager = (context as Context).getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    private lateinit var networkConnectivityCallbacks: ConnectivityCallbackNetworkManager
+    private lateinit var networkConnectivityReceiver: ConnectivityReceiverNetworkManager
+
+    private var _isNetworkEnabled = AtomicReference(false)
```

Comment:
`AtomicBoolean` but I question the need for this field, since it's already stored in the state machine.

Reply:
What do you mean "it's already stored in the state machine"

Reply:
There is a state repo that tells you if the network is enabled. There is also this field. Why have two?

On top of that as discussed elsewhere there is a third state (unknown), so we should drop this field.

Reply:
same as elsewhere, it's a duplication.

Reply:
853749a

Reply:
Yeah that reduced the code by a lot 👍